### PR TITLE
Automatically determine attenuation band

### PR DESCRIPTION
**Pull request by kuangdai**
From a user's perspective, automatic determination of f_min and f_max 
does provide a big convenience. This should work for both 2D and 3D.

### CODE REVIEW

File: pymesher/unstructured_mesh.py [hunk 1]
```diff
@@ -745,6 +745,29 @@ def compute_dt(self, vp, courant_number, min_gll_point_distance=1.):
         dt = float(dt_elem.min())
 
         return dt, dt_elem
+        
+    def compute_resolved_period(self, vmin, elements_per_wavelength):
+        """
+        estimate the the largest possible minimum period resolved (hmax / vmin).
```
Comment (martinvandriel) [inline pymesher/unstructured_mesh.py]: do you mean shortest resolved period?

Comment (martinvandriel) [inline pymesher/unstructured_mesh.py]: Now that I think about it: it might be useful as a quality measure of the mesh...

Comment (kuangdai) [inline pymesher/unstructured_mesh.py]: Hi, 

Code format has just been fixed.

Yes, the shortest resolved period, but the maximum of it among elements. Ideally, this should be a little smaller than -period.

Best,
Kuangdai

File: pymesher/attenuation.py [hunk 2]
```diff
@@ -171,3 +171,22 @@ def plot(self, ffac=10., nfsamp=1000, errorlim=1.1, show=True, color='r',
...
+        if nsls == 2:
+            return 0.75
+        if nsls == 3:
+            return 1.75
+        if nsls == 4:
+            return 2.25
+        if nsls == 5:
+            return 2.85
+        # otherwise use linear extrapolation based on 4 and 5
+        return (2.85 - 2.25) * (nsls - 4) + 2.25
```
Comment (martinvandriel) [inline pymesher/attenuation.py]: shouldn't this read

`(2.85 - 2.25) * (nsls - 5) + 2.85`

??

Comment (martinvandriel) [inline pymesher/attenuation.py]: I am adapting this to our method of finding the attenuation parameters following figure 4 in my attenuation paper

Comment (martinvandriel) [inline pymesher/attenuation.py]: my bad, the two formulas yield the same result

**Comment by martinvandriel** [comment]
I think this can be merged, if @kuangdai is ok with my changes.

**Pull request merged**